From 03efe234974d2511b04dc3733b21261a50fa6941 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 27 Jun 2023 11:19:47 +0200
Subject: [PATCH] execute: don't bother with chowning StateDirectory= and
 friends in user mode

(cherry picked from commit f5bb36dcfe71dab3f79e8e6133a2f4260d91f213)

Related: RHEL-137252
---
 src/core/execute.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/core/execute.c b/src/core/execute.c
index 0ae4640382..9e3da5315f 100644
--- a/src/core/execute.c
+++ b/src/core/execute.c
@@ -2570,6 +2570,11 @@ static int setup_exec_directory(
                 if (r < 0)
                         goto fail;
 
+                /* Skip the rest (which deals with ownership) in user mode, since ownership changes are not
+                 * available to user code anyway */
+                if (params->runtime_scope != RUNTIME_SCOPE_SYSTEM)
+                        continue;
+
                 /* Then, change the ownership of the whole tree, if necessary. When dynamic users are used we
                  * drop the suid/sgid bits, since we really don't want SUID/SGID files for dynamic UID/GID
                  * assignments to exist. */
